src/main/java/uk/gov/ons/ctp/response/action/service/decorator/context/BusinessActionRequestContextFactory.java

```diff
@@ -6,36 +6,63 @@
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Collectors;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.response.action.client.CaseSvcClientService;
+import uk.gov.ons.ctp.response.action.client.CollectionExerciseClientService;
 import uk.gov.ons.ctp.response.action.client.PartySvcClientService;
+import uk.gov.ons.ctp.response.action.client.SurveySvcClientService;
 import uk.gov.ons.ctp.response.action.domain.model.Action;
+import uk.gov.ons.ctp.response.action.domain.model.ActionPlan;
+import uk.gov.ons.ctp.response.action.domain.repository.ActionPlanRepository;
+import uk.gov.ons.ctp.response.lib.casesvc.representation.CaseDetailsDTO;
+import uk.gov.ons.ctp.response.lib.casesvc.representation.CaseGroupDTO;
+import uk.gov.ons.ctp.response.lib.collection.exercise.representation.CollectionExerciseDTO;
 import uk.gov.ons.ctp.response.lib.party.representation.Association;
 import uk.gov.ons.ctp.response.lib.party.representation.PartyDTO;
 import uk.gov.ons.ctp.response.lib.sample.representation.SampleUnitDTO.SampleUnitType;
+import uk.gov.ons.ctp.response.lib.survey.representation.SurveyDTO;
 
 @Component
-@Qualifier("business")
 public class BusinessActionRequestContextFactory implements ActionRequestContextFactory {
   private static final Logger log =
       LoggerFactory.getLogger(BusinessActionRequestContextFactory.class);
   private static final String SAMPLE_UNIT_TYPE_NOT_SUPPORTED =
       "Only sample unit type B supported, sampleUnitType=%s not supported";
 
   private final PartySvcClientService partySvcClientService;
-
-  private final DefaultActionRequestContextFactory defaultFactory;
+  private final ActionPlanRepository actionPlanRepo;
+  private final CaseSvcClientService caseSvcClientService;
+  private final CollectionExerciseClientService collectionExerciseClientService;
+  private final SurveySvcClientService surveySvcClientService;
 
   public BusinessActionRequestContextFactory(
       PartySvcClientService partySvcClientService,
-      DefaultActionRequestContextFactory defaultFactory) {
+      ActionPlanRepository actionPlanRepo,
+      CaseSvcClientService caseSvcClientService,
+      CollectionExerciseClientService collectionExerciseClientService,
+      SurveySvcClientService surveySvcClientService) {
     this.partySvcClientService = partySvcClientService;
-    this.defaultFactory = defaultFactory;
+    this.actionPlanRepo = actionPlanRepo;
+    this.caseSvcClientService = caseSvcClientService;
+    this.collectionExerciseClientService = collectionExerciseClientService;
+    this.surveySvcClientService = surveySvcClientService;
   }
 
   @Override
   public ActionRequestContext getActionRequestDecoratorContext(Action action) {
```

Comment:
I don't expect this to be fixed here, but this remains an absurdly inefficient method that's almost singlehandedly responsible for this part of the system being non-performant - we should scrap it as soon as we can.

Reply:
100% percentage agree. I actually don't like this decorator pattern here at all. Let's get rid of it on the next pass.